scripts/core/editor3/components/Editor3Component.tsx

```diff
@@ -645,85 +676,98 @@ export class Editor3Component extends React.Component<IPropsEditor3Component, IS
-                        onBlur={(event: any) => {
-                            if (
-                                event?.relatedTarget == null
+                                const selectionRect = getVisibleSelectionRect(window);
+
+                                if (this.editorNode?.current != null && selectionRect != null) {
+                                    this.editorNode.current.dataset.editorSelectionRect = JSON.stringify(selectionRect);
+                                }
+
+                                onChange(editorStateNext);
+                            }}
+                            tabIndex={tabindex}
+                            handlePastedText={handlePastedText.bind(this)}
+                            readOnly={locked || readOnly}
+                            ref={(editor) => this.handleRefs(editor)}
+                            spellCheck={appConfig.editor3.browserSpellCheck}
+                            stripPastedStyles={cleanPastedHtml}
+                            onBlur={(event: any) => {
+                                if (event?.relatedTarget == null
                                 || querySelectorParent(
                                     event.relatedTarget,
                                     `.${editor3AutocompleteClassName}`,
                                     {self: true},
-                                ) == null
-                            ) { // check whether focus went to autocomplete or to an item inside it
-                                this.setState({contentChangesAfterLastFocus: 0});
-                            }
-                        }}
-                    />
+                                ) == null) { // check whether focus went to autocomplete or to an item inside it
+                                    this.setState({contentChangesAfterLastFocus: 0});
+                                }
+                            }}
+                        />
 
-                    {
-                        this.state.contentChangesAfterLastFocus > 0 && (
+                        {this.state.contentChangesAfterLastFocus > 0 && (
                             <Editor3Autocomplete
                                 editorState={editorState}
                                 editorNode={this.editorNode.current}
                                 dispatch={this.props.dispatch}
                                 autocompleteSuggestions={this.props.autocompleteSuggestions}
                                 className={editor3AutocompleteClassName}
                             />
-                        )
-                    }
+                        )}
 
-                    {this.props.loading && <div className="loading-overlay active" />}
+                        {this.props.loading && <div className="loading-overlay active" />}
+                    </div>
                 </div>
-            </div>
+                {this.showExpandButton && this.props.expandable?.enabled && (
```

Comment:
If you are doing conditional rendering based on a value - you probably want things to re-render if that value changes - thus you should store it in the state rather than as a component property.